### PR TITLE
feat: add docs for offline form submission

### PR DESCRIPTION
**Pull request by haijian-vaadin**
Add documentation for deferrable endpoint requests
For #1339

### CODE REVIEW

File: documentation/pwa/tutorial-offline-form-submission.asciidoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+title: Offline Form Submission
+order: 8
+layout: page
+---
+
+= Offline Form Submission
+
+Our TypeScript <<../client-side-forms/tutorial-binder#,Binder>> works seamlessly with the <<tutorial-defer-endpoint-request#, deferrable endpoints>>, which allows your form submission to work both online and offline. 
+
+When the application is online, the form will be submitted normally. When offline, the form will be cached locally in the browser and be submitted automatically when coming back online.
```
Comment (platosha) [inline documentation/pwa/tutorial-offline-form-submission.asciidoc:11]: “Caching” typically means additional storage, while here it is the only one storage for the offline period of time. Let us rather say ”stored”.

```suggestion
When the application is online, the form is submitted normally. When offline, the form is stored locally in the browser and is submitted automatically when coming back online.
```

**Review by platosha — CHANGES_REQUESTED** [review]

File: documentation/pwa/tutorial-offline-form-submission.asciidoc (hunk 1 above)
Comment (haijian-vaadin) [reply documentation/pwa/tutorial-offline-form-submission.asciidoc]: done

File: documentation/pwa/tutorial-defer-endpoint-request copy.asciidoc [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+When creating UI with TypeScript, all the logic is done on the client-side, which means that they work offline. But not the Endpoint requests, which requires network access by default.
+
+Vaadin stores the Endpoint requests locally in the browser when the application is offline. The deferred requests will be submitted automatically when the application is back online.
+
+== @Deferrable annotation
+
+You can make an endpoint deferrable by adding a `@Deferrable` annotation. It works either on the Endpoint class or a single method.
+
+*Example*: Using the `@Deferrable` annotation to defer endpoints.
```
[the remark this answers is not in the document]
Comment (haijian-vaadin) [reply documentation/pwa/tutorial-defer-endpoint-request copy.asciidoc]: done

**Comment by haijian-vaadin** [comment]
Thanks for the good review, applied the suggestions

**Review by platosha — APPROVED** [review]

**Pull request merged**